File: PropertyGrids/UserPreferencesExample/UserPreferencesExample/UserPreferencesExample/UserPreferenceSample.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using UserPreferencesExample.Properties;

namespace UserPreferencesExample
{
    public partial class UserPreferenceSample : Form
    {
        Settings defaultSettings;
        public UserPreferenceSample()
        {
            InitializeComponent();
        }

        private void btnSave_Click(object sender, EventArgs e)
        {
            if (defaultSettings != null)
            {
                //Save button's settings
                defaultSettings.ButtonBackColor = btnSave.BackColor;
                defaultSettings.SaveButtonText = btnSave.Text;

                //Save the Form's Settings
                defaultSettings.FirstFormHeader = this.Text;
                defaultSettings.FormBackColor = this.BackColor;
                defaultSettings.FormForeColor = this.ForeColor;
                defaultSettings.FormOpacity = this.Opacity;
                defaultSettings.FormWindowState = this.WindowState;

                defaultSettings.Save();
            }
        }

        private void MyFirstForm_Load(object sender, EventArgs e)
        {
            defaultSettings = new Settings();
        }

        private void txtFormHeader_KeyPress(object sender, KeyPressEventArgs e)
        {
            
            if (this.Text.Equals(defaultSettings.FirstFormHeader))
                this.Text = string.Empty;
            else
                this.Text = this.txtFormHeader.Text;
        }

        private void txtButtonText_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (this.btnSave.Text.Equals(defaultSettings.FirstFormHeader))
                this.btnSave.Text = string.Empty;
            else
                this.btnSave.Text = this.txtButtonText.Text;
        }
    }
}

File: PropertyGrids/UserPreferencesExample/UserPreferencesExample/UserPreferencesExample.ConvertedToVBNet/UserPreferenceSample.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports UserPreferencesExample.Properties

Public Partial Class UserPreferenceSample
	Inherits Form
	Private defaultSettings As Settings
	Public Sub New()
		InitializeComponent()
	End Sub

	Private Sub btnSave_Click(sender As Object, e As EventArgs)
		If defaultSettings IsNot Nothing Then
			'Save button's settings
			defaultSettings.ButtonBackColor = btnSave.BackColor
			defaultSettings.SaveButtonText = btnSave.Text

			'Save the Form's Settings
			defaultSettings.FirstFormHeader = Me.Text
			defaultSettings.FormBackColor = Me.BackColor
			defaultSettings.FormForeColor = Me.ForeColor
			defaultSettings.FormOpacity = Me.Opacity
			defaultSettings.FormWindowState = Me.WindowState
			defaultSettings.Save()
		End If
	End Sub

	Private Sub MyFirstForm_Load(sender As Object, e As EventArgs)
		defaultSettings = New Settings()
	End Sub

	Private Sub txtFormHeader_KeyPress(sender As Object, e As KeyPressEventArgs)

		If Me.Text.Equals(defaultSettings.FirstFormHeader) Then
			Me.Text = String.Empty
		Else
			Me.Text = Me.txtFormHeader.Text
		End If
	End Sub

	Private Sub txtButtonText_KeyPress(sender As Object, e As KeyPressEventArgs)
		If Me.btnSave.Text.Equals(defaultSettings.FirstFormHeader) Then
			Me.btnSave.Text = String.Empty
		Else
			Me.btnSave.Text = Me.txtButtonText.Text
		End If
	End Sub
End Class
